sssom/util.py

```diff
@@ -143,22 +143,26 @@ def __str__(self) -> str:  # noqa:D105
             description += self.df.tail().to_string() + "\n"
         return description
 
-    def clean_prefix_map(self) -> None:
+    def clean_prefix_map_and_metadata(self) -> None:
```

Comment:
Sorry I think I explained this wrong. The function should only clean the prefix_map, but _take into account any prefixes used in the metadata_. So I expected something like:

```
prefixes_in_map = get_prefixes_used_in_table(self.df)
prefixes_in_metadata = get_prefixes_used_in_metadata(self.metadata)
prefixes_in_msdf = prefixes_in_map
prefixes_in_msdf.extent(prefixes_in_metadata)
```

Or some such? So: make sure you don't remove prefixes from the prefix map that are needed for the metadata!

Reply:
Does the new solution make sense?

Reply:
This is what I am thinking with:

```
prefixes_in_metadata = get_prefixes_used_in_metadata(self.metadata)
```

This is example metadata:

```
#mapping_set_id: MGI_KidsFirst_MP_HPO
#mapping_set_title: Mapping of MP terms to HPO terms from KidsFirst list
#mapping_set_description: "HPO terms were provided by the KidsFirst team from the set of terms used for patients in their cohorts. Initially the mapping date was not tracked so these were given the date of when we recieved the file (5/27/21). The set of HPO terms was updated on 5/17/212. Use of skos mappings terms: exact match: MP term is the same as the HPO term, ignoring that the HP term applies only to humans; broad match: MP term covers more than the HPO term; narrow match: MP term is more restircted than the HPO term, ignoring the species restriction; related match: MP term and HPO term share some similarity but would not have a parent-child relation in a shared ontology"
#creator_id: orcid:0000-0003-4606-0597|orcid:0000-0002-6490-7723|ror:021sy4w91|wikidata:Q1951035
#license: https://creativecommons.org/licenses/by/4.0/
#object_source: obo:hp
#subject_source: obo:mp
```

The prefixes are not `object_source` or `creator_id` as your current code seems to suggest. The prefixes are `obo:` as used in the `object_source` field, and `orcid:` as used in the `creator_id` field. So to get all the prefixes in the metadata, you have to loop through the values of the metadata (be they literals or lists!) and make sure to extract all the prefixes used in there - not in the keys!

Reply:
Done!